Index: mt76-2025.11.06~eb567bc7/mt76_connac_mcu.c
===================================================================
--- mt76-2025.11.06~eb567bc7.orig/mt76_connac_mcu.c
+++ mt76-2025.11.06~eb567bc7/mt76_connac_mcu.c
@@ -1735,7 +1735,7 @@ int mt76_connac_mcu_hw_scan(struct mt76_
 {
 	struct mt76_vif_link *mvif = (struct mt76_vif_link *)vif->drv_priv;
 	struct cfg80211_scan_request *sreq = &scan_req->req;
-	int n_ssids = 0, err, i, duration;
+	int n_ssids = 0, err, i;
 	int ext_channels_num = max_t(int, sreq->n_channels - 32, 0);
 	struct ieee80211_channel **scan_list = sreq->channels;
 	struct mt76_dev *mdev = phy->dev;
@@ -1774,14 +1774,6 @@ int mt76_connac_mcu_hw_scan(struct mt76_
 	req->ssid_type_ext = n_ssids ? BIT(0) : 0;
 	req->ssids_num = n_ssids;
 
-	duration = is_mt7921(phy->dev) ? 0 : MT76_CONNAC_SCAN_CHANNEL_TIME;
-	/* increase channel time for passive scan */
-	if (!sreq->n_ssids)
-		duration *= 2;
-	req->timeout_value = cpu_to_le16(sreq->n_channels * duration);
-	req->channel_min_dwell_time = cpu_to_le16(duration);
-	req->channel_dwell_time = cpu_to_le16(duration);
-
 	if (sreq->n_channels == 0 || sreq->n_channels > 64) {
 		req->channel_type = 0;
 		req->channels_num = 0;
