From 1d602fe0fb2a7fb132a1813fdd81e90bd9077de0 Mon Sep 17 00:00:00 2001
From: Ruslan Isaev <legale.legale@gmail.com>
Date: Wed, 6 May 2026 01:14:52 +0300
Subject: [PATCH] uboot-mediatek: l2sh: fixup l2sh_proto.h compilation warning

Signed-off-by: Ruslan Isaev <legale.legale@gmail.com>
---
 include/l2sh_proto.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/l2sh_proto.h b/include/l2sh_proto.h
index 4f3c95679..a4516dd20 100644
--- a/include/l2sh_proto.h
+++ b/include/l2sh_proto.h
@@ -187,7 +187,7 @@ typedef struct l2s_arq_view {
 static const u8 l2s_shared_key[4] = {0x4a, 0x17, 0x59, 0xc3};
 
 static inline u16 l2s_get_be16(const u8 *buf) {
-  return ((u16)buf[0] << 8) | (u16)buf[1];
+  return (u16)((buf[0] << 8) | buf[1]);
 }
 
 static inline u32 l2s_get_be32(const u8 *buf) {
-- 
2.39.5

