Index: strace-6.14/configure.ac
===================================================================
--- strace-6.14.orig/configure.ac
+++ strace-6.14/configure.ac
@@ -36,6 +36,7 @@ AC_SUBST(RPM_CHANGELOGTIME)
 DEB_CHANGELOGTIME="$(LC_TIME=C date -u -R)"
 AC_SUBST(DEB_CHANGELOGTIME)
 
+m4_include([m4/ac_prog_cc_override.m4])
 AC_PROG_CC
 AC_PROG_CPP
 AC_PROG_INSTALL
Index: strace-6.14/m4/ac_prog_cc_override.m4
===================================================================
--- strace-6.14.orig/m4/ac_prog_cc_override.m4
+++ strace-6.14/m4/ac_prog_cc_override.m4
@@ -8,3 +8,4 @@ AC_DEFUN([AC_PROG_CC],
                  [AC_MSG_RESULT([yes])],
                  [AC_MSG_ERROR([compiler does not accept -std=gnu11])])
 ])
+
