Index: src/Makefile
===================================================================
--- src.orig/Makefile
+++ src/Makefile
@@ -1098,11 +1098,11 @@ AUTOMAKE = ${SHELL} '/mnt/2tb/sdk4/ax840
 AWK = gawk
 BUILD_EXEEXT = 
 BUILD_OBJEXT = 
-CC = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23
+CC = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x
 CCDEPMODE = depmode=none
-CC_FOR_BUILD = /mnt/2tb/sdk4/ax840-ap-master/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq60xx/strace-6.14/build-aux/compile gcc -std=gnu23
+CC_FOR_BUILD = /mnt/2tb/sdk4/ax840-ap-master/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq60xx/strace-6.14/build-aux/compile gcc -std=gnu2x
 CC_FOR_M32 = 
-CC_FOR_MX32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23
+CC_FOR_MX32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x
 CFLAGS = -Os -pipe -mcpu=cortex-a53 -fno-caller-saves -fno-plt -fhonour-copts -fmacro-prefix-map=/mnt/2tb/sdk4/ax840-ap-master/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq60xx/strace-6.14=strace-6.14 -Wformat -Werror=format-security -fstack-protector -D_FORTIFY_SOURCE=1 -Wl,-z,now -Wl,-z,relro   
 CFLAGS_FOR_BUILD = -O2 -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/host/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/hostpkg/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/target-aarch64_cortex-a53_musl/host/include -I/mnt/2tb/sdk4/ax840-ap-master/build_dir/target-aarch64_cortex-a53_musl/linux-qualcommax_ipq60xx/linux-6.12.40/user_headers/include
 CFLAGS_FOR_M32 = 
@@ -1114,14 +1114,14 @@ CODE_COVERAGE_ENABLED = no
 CODE_COVERAGE_LDFLAGS = 
 CODE_COVERAGE_LIBS = 
 COPYRIGHT_YEAR = 2025
-CPP = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23 -E
+CPP = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x -E
 CPPFLAGS = -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/usr/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include/fortify    -isystem ./bundled/linux/arch/arm64/include/uapi -isystem ./bundled/linux/include/uapi
 CPPFLAGS_FOR_BUILD = -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/host/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/hostpkg/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/target-aarch64_cortex-a53_musl/host/include
 CPPFLAGS_FOR_M32 = -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/usr/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include/fortify    -isystem ./bundled/linux/arch/arm64/include/uapi -isystem ./bundled/linux/include/uapi
 CPPFLAGS_FOR_MX32 = -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/usr/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include/fortify    -isystem ./bundled/linux/arch/arm64/include/uapi -isystem ./bundled/linux/include/uapi
 CPP_FOR_BUILD = cpp
-CPP_FOR_M32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23 -E
-CPP_FOR_MX32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23 -E
+CPP_FOR_M32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x -E
+CPP_FOR_MX32 = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x -E
 CSCOPE = cscope
 CTAGS = ctags
 CYGPATH_W = echo
@@ -11303,7 +11303,7 @@ native_printer_decls.h native_printer_de
 #$(mpers_mx32_targets): mpers_NAME = mx32
 #$(mpers_mx32_targets): mpers_CC_FLAGS =  -mx32
 #$(mpers_mx32_targets): mpers_CPPFLAGS = -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/usr/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include -I/mnt/2tb/sdk4/ax840-ap-master/staging_dir/toolchain-aarch64_cortex-a53_gcc-14.3.0_musl/include/fortify    -isystem ./bundled/linux/arch/arm64/include/uapi -isystem ./bundled/linux/include/uapi
-#$(mpers_mx32_targets): mpers_CC = ccache aarch64-openwrt-linux-musl-gcc -std=gnu23
+#$(mpers_mx32_targets): mpers_CC = ccache aarch64-openwrt-linux-musl-gcc -std=gnu2x
 
 clean-local:
 	-rm -rf mpers-m32 mpers-mx32
